### DISCUSSION TITLE
Hook with item.id empty

### DISCUSSION THREAD

**Discussion by Arthur-Ruthenberg** [keystonejs/keystone#9403] [Questions]
Hi,
I had a. very unusual behavior the other day. 
On one of my lists I have an afterOperation hook that performs some stuff on UPDATE operation, but somehow item.id came as undefined.
It happened only once, and the code is in production and it never faulted before. 
Anyone happened to see something like this?

**Comment by gautamsi** [comment]
I have seen this, unable to reproduce consistently.

**Reply by Arthur-Ruthenberg** [reply]
Any ideas how or why this might happen?
This is very very serious bug.

**Reply by dcousens** [reply]
@Arthur-Ruthenberg can you provide any details about the list in question? What is your `idField` configuration?

**Reply by Arthur-Ruthenberg** [reply]
@dcousens Hi
They id configuration is out of the box. We don't change it.
But I do can tell you that the incident happened on an afterOperation hook on Update operation.
A user tried to update 2 fields, one unique and one not. There was an update check that has failed due to unique constrains, but the afterOperation still activated even though the use received an error
